omegatrade-app/backend/app/controller/user.controller.js

```diff
@@ -0,0 +1,166 @@
+            }
+            const payload = {
+                'userId': user.userId,
+                'fullName': user.fullName,
+                'businessEmail': user.businessEmail,
+                'photoUrl': params.photoUrl,
+                'provider': params.provider
+            };
+            jwt.sign(payload, process.env.JWT_SECRET, {
+                expiresIn: process.env.EXPIRE_IN
+            }, function (err, token) {
+                if (err) {
+                    res.status(400).json({
+                        success: false,
+                        message: 'Something went wrong while registering new user!'
+                    });
+                }
+                res.json({
+                    success: true,
+                    message: 'Registered successfully!',
+                    userInfo: payload,
+                    authToken: token
+                });
+            });
+        });
+    } catch (err) {
+        res.status(400).json({
+            success: false,
+            message: err
+        })
+    }
+};
+
+/**
+ * Function to validate and login user.
+ * @method POST
+ * 
+ * @param {*} req contains request headers and payload businessEmail,password
+ */
+exports.login = function (req, res) {
+    const params = req.body;
+    User.findUser(params, function (err, user) {
+        if (user && user.userId) {
+            if (bcrypt.compareSync(params.password, user.password)) {
+                if (user) {
+                    const payload = {
+                        'userId': user.userId,
+                        'fullName': user.fullName,
+                        'businessEmail': user.businessEmail,
+                        'photoUrl': user.photoUrl,
+                        'provider': user.provider
+                    };
+                    const token = jwt.sign(payload, process.env.JWT_SECRET, {
+                        expiresIn: process.env.EXPIRE_IN
+                    });
+                    res.status(200).json({
+                        success: true,
+                        message: 'Logged in successfully',
+                        userInfo: payload,
+                        authToken: token
+                    });
+                }
+            } else
+                res.status(401).json({
+                    success: false,
+                    message: 'invalid password!'
+                });
+        } else
+            res.status(401).json({
+                success: false,
+                message: 'invalid username or password'
+            });
+        if (err)
+            res.send(err);
+    });
+};
+
+/**
+ * Function to generate JWT token
+ * @method POST
+ * 
+ * @param {*} req contains request headers and payload businessEmail,password
+ */
+exports.getTokenSocial = function (req, res) {
+    const params = req.body;
+    if (params && params.provider === 'GOOGLE') {
+        User.findUser({
+            'businessEmail': params.email
+        }, function (err, user) {
```

Comment:
findUser needs consistency here. either pass params or use like this passing only businessEmail property. also why businessEmail is wrap into single quote?